tests/system/data/test_system_async.py

```diff
@@ -1116,6 +1122,7 @@ async def test_execute_query_params(self, client, table_id, instance_id):
             "tsArrayParam": SqlType.Array(SqlType.Timestamp()),
             "dateArrayParam": SqlType.Array(SqlType.Date()),
         }
+
         result = await client.execute_query(
             query, instance_id, parameters=parameters, parameter_types=param_types
         )
```

Comment:
are there any new system tests that can be added? Maybe using the new `prepare_*` arguments? Or accessing `iterator.metadata`?

Reply:
I want to add at least one against a real table, and I want to refactor the Metadata to expose the fields in ProtoMetadata. I was planning both of those for a follow on if that's ok (still to the branch before we merge).

For prepare_* it's a bit tricky since those are all retry/timeout params